### ISSUE TITLE
Error with NbPasswordAuthStrategy on v4 and target es2015

### ISSUE THREAD

**Issue by ov3rk1ll** [akveo/nebular#1565]
### Issue type

**I'm submitting a ...**  (check one with "x")

* [x] bug report
* [ ] feature request

### Issue description

**Current behavior:**
When using NbPasswordAuthStrategy with target es2015, the site crashes with error
```
Error: Uncaught (in promise): ReferenceError: Must call super constructor in derived class before accessing 'this' or returning from derived constructor
ReferenceError: Must call super constructor in derived class before accessing 'this' or returning from derived constructor
    at new gt (index.js.pre-build-optimizer.js:1737)
    at core.js.pre-build-optimizer.js:23174
    at bf (core.js.pre-build-optimizer.js:23137)
    at yf (core.js.pre-build-optimizer.js:23084)
    at Hf.get (core.js.pre-build-optimizer.js:24192)
    at index.js.pre-build-optimizer.js:2306
    at Array.forEach (<anonymous>)
    at zt (index.js.pre-build-optimizer.js:2305)
    at core.js.pre-build-optimizer.js:23201
    at bf (core.js.pre-build-optimizer.js:23140)
    at yf (core.js.pre-build-optimizer.js:23084)
    at core.js.pre-build-optimizer.js:23199
    at bf (core.js.pre-build-optimizer.js:23140)
    at yf (core.js.pre-build-optimizer.js:23084)
    at core.js.pre-build-optimizer.js:23174
    at bf (core.js.pre-build-optimizer.js:23137)
    at yf (core.js.pre-build-optimizer.js:23084)
    at core.js.pre-build-optimizer.js:23172
    at bf (core.js.pre-build-optimizer.js:23137)
    at yf (core.js.pre-build-optimizer.js:23084)
    at core.js.pre-build-optimizer.js:23172
    at bf (core.js.pre-build-optimizer.js:23137)
    at yf (core.js.pre-build-optimizer.js:23084)
    at core.js.pre-build-optimizer.js:23174
    at bf (core.js.pre-build-optimizer.js:23137)
    at yf (core.js.pre-build-optimizer.js:23084)
    at core.js.pre-build-optimizer.js:23176
    at bf (core.js.pre-build-optimizer.js:23137)
    at yf (core.js.pre-build-optimizer.js:23084)
    at core.js.pre-build-optimizer.js:23174
    at bf (core.js.pre-build-optimizer.js:23137)
    at yf (core.js.pre-build-optimizer.js:23084)
    at Hf.get (core.js.pre-build-optimizer.js:24192)
    at Te (router.js.pre-build-optimizer.js:4126)
    at router.js.pre-build-optimizer.js:4514
    at t._subscribe (defer.js.pre-build-optimizer.js:8)
    at t._trySubscribe (Observable.js.pre-build-optimizer.js:42)
    at t.subscribe (Observable.js.pre-build-optimizer.js:28)
    at s.call (take.js.pre-build-optimizer.js:22)
    at t.subscribe (Observable.js.pre-build-optimizer.js:23)
    at a (subscribeToResult.js.pre-build-optimizer.js:9)
    at u._innerSub (mergeMap.js.pre-build-optimizer.js:59)
    at u._tryNext (mergeMap.js.pre-build-optimizer.js:53)
    at u._next (mergeMap.js.pre-build-optimizer.js:36)
    at u.next (Subscriber.js.pre-build-optimizer.js:49)
    at t._subscribe (subscribeToArray.js.pre-build-optimizer.js:3)
    at t._trySubscribe (Observable.js.pre-build-optimizer.js:42)
    at t.subscribe (Observable.js.pre-build-optimizer.js:28)
    at call (mergeMap.js.pre-build-optimizer.js:21)
    at t.subscribe (Observable.js.pre-build-optimizer.js:23)
    at P (zone-evergreen.js.pre-build-optimizer.js:797)
    at P (zone-evergreen.js.pre-build-optimizer.js:754)
    at zone-evergreen.js.pre-build-optimizer.js:858
    at a.invokeTask (zone-evergreen.js.pre-build-optimizer.js:391)
    at Object.onInvokeTask (core.js.pre-build-optimizer.js:30873)
    at a.invokeTask (zone-evergreen.js.pre-build-optimizer.js:390)
    at s.runTask (zone-evergreen.js.pre-build-optimizer.js:168)
    at _ (zone-evergreen.js.pre-build-optimizer.js:559)
(anonymous) @ breadcrumbs.js.pre-build-optimizer.js:58
```

(The index in the first line maps to 
webpack:///./node_modules/@nebular/auth/fesm2015/index.js.pre-build-optimizer.js)

When the target is set to es5, the same build works without any issues. "module" is "esnext" in both cases.

The project is based off ngx-admin with very little changes to any of the config/json files.

### Other information:

**npm, node, OS, Browser**
```
Node: v12.3.1, npm: 6.9.0
OS: Windows 10
Browser: Chrome and Firefox
```

**Angular, Nebular**
```
"@angular/common": "8.0.0",
"@nebular/auth": "4.0.0",
"@nebular/bootstrap": "4.0.0",
"@nebular/eva-icons": "4.0.0",
"@nebular/security": "4.0.0",
"@nebular/theme": "4.0.0",
```

**Comment by yggg** [comment]
Hi @ov3rk1ll, thanks for reporting! Just tried auth with a new angular app and `NbPasswordAuthStrategy` works fine for me. I've also checked class constructors in auth classes and looks like all of them call `super` first.
Could you please post code/screenshot of the class/constructor where the error originates? Also please share details on how are you running app (flags, etc.).

**Comment by ov3rk1ll** [comment]
Thank your for looking into this.

I'm building with `ng build --prod --base-href=/dist/ --sourceMap=true`. After that, I run `http-server` and open http://127.0.0.1:8080/dist/#/ in the browser. Same happens with `ng serve --prod`

The error in the developer console points to the line shown below:
![image](https://user-images.githubusercontent.com/1522953/58856409-e6695f00-86a2-11e9-9734-580e6e077f5f.png)
This has the super call so I honestly have no idea what is going on.

angular.json, tsconfig.json and tsconfig.app.json all match the content from https://github.com/akveo/ngx-admin/tree/feat/update-to-8.

I've removed package-lock.json and run npm reinstall. The issue does still occur.
I did notice that @angular-devkit/core was installed as 7.2.4 but I fixed that to 8.0.1 as well (it showed a error with some types in @nebular/theme which are gone now).
